cannon/mipsevm/memory/memory.go

```diff
@@ -98,8 +98,9 @@ func (m *Memory) invalidate(addr Word) {
 		return
 	}
 
-	// find the gindex of the first page covering the address
-	gindex := ((uint64(1) << 32) | uint64(addr)) >> PageAddrSize
+	// find the gindex of the first page covering the address: i.e. ((1 << WordSize) | addr) >> PageAddrSize
+	// Avoid 64-bit overflow by expanding the rsh over the OR.
```

Comment:
(nit) Had to stare at this comment a bit - maybe update to:
```suggestion
	// Avoid 64-bit overflow by distributing the right shift across the OR.
```